lib/utils/Thing.js

```diff
@@ -202,12 +204,36 @@ export default class Thing {
 		}
 	}
 
-	getExpandoButton() {
-		return this.entry.querySelector('.expando-button, .search-expando-button');
+	getEntryExpando() {
+		const button = $(this.entry).find('> .expando-button')[0];
+		if (!button) return null;
+
+		let expando = expandos.get(button);
+
+		if (!expando) {
+			// Otherwise it is a native expando: Create a pseudo-class for those
+			const isOpen = () => button.classList.contains('expanded');
+			const toggle = () => { click(button); };
+			expando = {
+				button,
+				get open() { return isOpen(); },
+				collapse() { if (isOpen()) toggle(); },
+				expand() { if (!isOpen()) toggle(); },
+			};
+		}
+
+		return expando;
+	}
+
+	getTextExpandos() {
+		let textExpandos = Array.from(this.entry.querySelectorAll(Thing.expandoSelector))
+			.map(v => expandos.get(v));
+		textExpandos = _.remove(textExpandos, this.getEntryExpando());
```

Comment:
`_.remove` operates in-place and returns an array of removed elements (i.e. just `this.getEntryExpando()`, if present); I'm not sure that that's what's intended here -- I would assume that this is intended to be:

```js
const entryExpando = this.getEntryExpando();
return Array.from(this.entry.querySelectorAll(Thing.expandoSelector))
    .map(v => expandos.get(v))
    .filter(v => v && v !== entryExpando);
```